src/server/api/routers/bot-source.ts

```diff
@@ -402,6 +405,101 @@ async function syncBotSourceSitemap(
     .where(eq(schema.botSources.id, bs.id))
 }
 
+function getSiteMaps() {
+  return publicProcedure
+    .input(
+      z.object({
+        url: z.string(),
+      }),
+    )
+    .query(async ({ input }) => {
+      // add https to url if not present
+      if (!input.url.startsWith('http')) {
+        input.url = `https://${input.url}`
+      }
+
+      // fetch robots.txt of the site
+      let robotsTxt
+
+      try {
+        robotsTxt = await fetch(`${input.url}/robots.txt`)
+      } catch (e) {
+        return []
+      }
+
+      const robotsTxtText = await robotsTxt.text()
+
+      // parse sitemap URLs from robots.txt
+      const sitemapUrls = parseSitemapUrls(robotsTxtText, input.url)
+      return sitemapUrls
+    })
+}
+
+function validateSitemapUrl() {
+  return publicProcedure
+    .input(
+      z.object({
+        url: z.string(),
+      }),
+    )
+    .query(async ({ input }) => {
+      return await isValidSitemapUrl(input.url)
+    })
+}
+
+export function parseSitemapUrls(
+  robotsTxt: string,
+  websiteUrl: string,
+): string[] {
+  const sitemapUrls: string[] = []
+  const lines = robotsTxt.split('\n')
+
+  for (const line of lines) {
+    const match = line.match(/Sitemap:\s*(.+)/i)
```

Comment:
This check function is also quite weird and manually.
We should use XMLParser to check and parse sitemap. You could take a look the code that parsed sitemap in this file as well.